src/ruby/puppetserver-lib/puppet/server/network/http/handler.rb

```diff
@@ -36,7 +36,25 @@ def path(request)
   def body(request)
     body = request["body"]
     if body.kind_of?(InputStream)
-      body.to_io.read()
+      io = body.to_io
+      content_type = ''
+      has_headers = request.key?('headers')
+      if has_headers && request['headers'].key?('content-type')
+         content_type = request['headers']['content-type']
```

Comment:
(nit) this is indented too much
```suggestion
        content_type = request['headers']['content-type']
```